### ISSUE TITLE
Dl4j not cleaning up temporary files

### ISSUE THREAD

**Issue by sorensf** [deeplearning4j/deeplearning4j#4833]
#### Issue Description
I noticed I had a lot of files "restore1254325221multilayer" (totalling about 80 gigabytes) in my temporary files folder (windows: C:\Users\<USERNAME>\AppData\Local\Temp linux (If i remember correctly): /var/tmp), which seems to be generated by dl4j whenever a network is loaded to memory.

The ones generated within a JVM is deleted if the JVM terminates nicely (e.g. the program finishes or an uncaught exception is thrown), but otherwise seem to stay forever. E.g. 
* If the JVM was killed externally or crashes the generated files stays forever, 
* If the server stays online indefinitely the files accumulate over time.

Expected behavior in my opinion would be that dl4j cleans up any temp files it doesn't actively use, or at the very least don't generate an unbounded amount of files.

#### Version Information
Dl4j/ND4j: Version 0.9.1
Platform: Tested on Windows 7 and Linux system
CUDA: 8.0 and CuDNN 6.0.
NVIDIA Driver: Latests on a GTX 1080.

**Comment by sorensf** [comment]
Source chat history: https://gist.github.com/sorensf/ec1307157b6bf55db786522eb5e30dc4#file-2018-03-23-14-30-17-30-chat-txt

**Comment by sorensf** [comment]
I would be very interested if there exists a "safe" workaround, e.g. run a service that deletes the temp folder every 5 minutes. But I'm not sure if the files are actively being used so this might break something.

**Comment by treo** [comment]
That problem is already fixed on master.

As a workaround, I think, you should be able to just copy the current version of https://github.com/deeplearning4j/deeplearning4j/blob/master/deeplearning4j-nn/src/main/java/org/deeplearning4j/util/ModelSerializer.java into your project, and use that.

**Comment by sorensf** [comment]
Ok, reading your comment I decided to read the two versions of ModelSerializer.java, and it seems that the main change is that there is a finally-block which attempts to delete the file - after the network has been fully restored. In my setup it was somewhat difficult to include the new version of ModelSerializer.java, so I decided to make the workaround in my code. Leaving this workaround here for anyone else interested: https://gist.github.com/sorensf/43f40b8b8aff6d3805b73c23ad327c05 
(Note the static lock so that multiple threads do not attempt to compete on deleting each others files.)

Since this is fixed on master I guess this issue should just be closed?
